wear/src/main/java/io/homeassistant/companion/android/sensors/HealthServicesSensorManager.kt

```diff
@@ -124,6 +218,61 @@ class HealthServicesSensorManager : SensorManager {
                 passiveListenerCallback
             )
         }
+
+        // Assume the callback is registered to avoid making multiple requests
         callBackRegistered = true
     }
+
+    private fun clearHealthServicesCallBack() {
+        passiveMonitoringClient?.clearPassiveListenerCallbackAsync()
+        callBackRegistered = false
+    }
+
+    private fun getActivityIcon(info: UserActivityInfo): String {
+        return when (info.userActivityState) {
+            UserActivityState.USER_ACTIVITY_EXERCISE -> {
+                when (info.exerciseInfo?.exerciseType) {
+                    ExerciseType.ALPINE_SKIING, ExerciseType.SKIING -> "mdi:skiing"
+                    ExerciseType.WEIGHTLIFTING, ExerciseType.BARBELL_SHOULDER_PRESS, ExerciseType.BENCH_PRESS -> "mdi:weight-lifter"
+                    ExerciseType.BIKING, ExerciseType.BIKING_STATIONARY, ExerciseType.MOUNTAIN_BIKING -> "mdi:bike"
+                    ExerciseType.SWIMMING_POOL, ExerciseType.SWIMMING_OPEN_WATER -> "mdi:swim"
+                    ExerciseType.BASEBALL -> "mdi:baseball"
+                    ExerciseType.BASKETBALL -> "mdi:basketball"
+                    ExerciseType.FOOTBALL_AMERICAN -> "mdi:football"
+                    ExerciseType.FOOTBALL_AUSTRALIAN -> "mdi:football-australian"
+                    ExerciseType.SOCCER -> "mdi:soccer"
+                    ExerciseType.SKATING, ExerciseType.INLINE_SKATING -> "mdi:skate"
+                    ExerciseType.ROLLER_SKATING -> "mdi:roller-skate"
+                    ExerciseType.SCUBA_DIVING -> "mdi:diving-scuba"

```

Comment:
This was [renamed in MDI 7.0.96](https://dev.materialdesignicons.com/changelog#version-7.0.96)

```suggestion
                    ExerciseType.SCUBA_DIVING -> "mdi:diving-scuba-mask"
```